envoy/stream_info/address_set_accessor.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+#include "envoy/network/address.h"
+#include "envoy/stream_info/filter_state.h"
+
+namespace Envoy {
+namespace StreamInfo {
+
+/**
+ * A FilterState object that holds a set of network addresses.
+ */
+class AddressSetAccessor : public FilterState::Object {
```

Comment:
I don't feel super strongly about it, but I'm not sure this abstraction is useful. I would just have a container that implements FilterState::Object and just have a set/list/etc. inside of it.

Reply:
Good point. will change it to something generic  like: 

```
template <typename T>
class SetAccessor : public FilterState::Object
```

or 

```
template <typename T>
class SetFilterState : public FilterState::Object

```